src/backend/api/handlers/insights.py

```diff
@@ -0,0 +1,42 @@
+from flask import Response
+
+from backend.api.handlers.decorators import api_authenticated, validate_keys
+from backend.api.handlers.helpers.profiled_jsonify import profiled_jsonify
+from backend.api.handlers.helpers.track_call import track_call_after_response
+from backend.common.consts.api_version import ApiMajorVersion
+from backend.common.decorators import cached_public
+from backend.common.models.insight import Insight
+from backend.common.queries.insight_query import InsightAllYearsQuery
+
+
+@api_authenticated
+@validate_keys
+@cached_public
+def insights_notables() -> Response:
+    """
+    Returns data about teams that may be:
+    - HOF
+    - CMP winners
+    - WFA winners (todo)
+    """
+    track_call_after_response("insights/notables")
+    hof = {
+        insight_dict["year"]: insight_dict["data_json"]
+        for insight_dict in InsightAllYearsQuery(
```

Comment:
`InsightAllYearsQuery`, as implemented, queries for all year-specific insights, not the aggregated insights across all years. Is that what you want?